sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -326,6 +336,32 @@ def delete_entity(
                 :dedent: 12
                 :caption: Deleting an entity of a Table
         """
+    
+    @overload
+    def delete_entity(
+        self,
+        entity: T,
+        *,
+        etag: Optional[str] = None,
+        match_condition: Optional[MatchConditions] = None,
+        encoder: TableEntityEncoderABC[T],
+        **kwargs: Any
+    ) -> None:
+        """Deletes the specified entity in a table. No error will be raised if
+        the entity or PartitionKey-RowKey pairing is not found.
+
+        :param entity: The entity to delete.
+        :type entity: Custom entity type
+        :keyword etag: Etag of the entity.
+        :paramtype etag: str or None
+        :keyword match_condition: The condition under which to perform the operation.
+            Supported values include: MatchConditions.IfNotModified, MatchConditions.Unconditionally.
+        :paramtype match_condition: ~azure.core.MatchConditions
+        :keyword encoder: The encoder used to serialize the outgoing Tables entities.
+        :type encoder: ~azure.data.Tables.TableEntityEncoderABC
+        :return: None
+        :raises: :class:`~azure.core.exceptions.HttpResponseError`
+        """
 
     @distributed_trace
     def delete_entity(self, *args: Union[EntityType, str], **kwargs: Any) -> None:
```

Comment:
The object `entity` could be an unknown object, so we need a way to access pk and rk without knowing what the object is. The only way I can think of for that is something like this....
```suggestion
    def delete_entity(self, *args: Union[EntityType, str, T], **kwargs: Any) -> None:
        encoder = kwargs.pop("encoder", DEFAULT_ENCODER)
        try:
            entity = kwargs.pop("entity", None)
            if not entity:
                entity = args[0]
            encoded = encoder.encode_entity(entity)
            partition_key = entity["PartitionKey"]
            row_key = entity["RowKey"]
        except:
            ...
        
```
Though this does result in a behaviour change.

Reply:
* After taking a closer look, I don't think we need to change as I don't see difference when getting encoder ahead.
* Do you mean get the value of pk and rk from encoded result? That's not necessary, as we are passing pk and rk values(not include odata type) and these values will encode by `prepare_key()`.

Reply:
Updated.